### ISSUE TITLE
Add binary transfer logic to KompactRef

### ISSUE THREAD

**Issue by Max-Meldrum** [cda-group/jobmanagement#15]
Will require implementation in kompact. Look to add as optional feature "file-transfer"? 

1. New FileTransfer spnl Frame?
2. Non-blocking tokio transfer handler
3. On completion, send msg to local actor with path to file.

**Comment by Bathtor** [comment]
What is this for?

**Comment by Max-Meldrum** [comment]
The idea is to simplify the binary transfer from the runtime to the execution layer. A rust process "compiler assistant" is spawned and checks llvm target triple, cpu flags etc.. it sends it off to the runtime using kompactakka proxy for compilation to happen. Then rather than creating more sockets or using some other library for the file transfer. One approach would be to run "kompactRef binaryTransfer path/bytes".

**Comment by Bathtor** [comment]
Aren't we anyway running a Scala management component on each host, that could be tasked with this?

**Comment by Max-Meldrum** [comment]
Yeah, this could be a Scala component. I guess I thought it would be fun to code some Rust and make something more lightweight (no jvm for docker, no large jar). But yeah, as you say, might be better to keep it as Scala and avoid adding additional logic to Kompact.

**Comment by Bathtor** [comment]
And also avoid the chicken and egg problem where the binary is supposed to fetch itself? :P

**Comment by Max-Meldrum** [comment]
One can always do an exec and replace the current process :p
